File: example_service_pkg/scripts/service_client.py
#!/usr/bin/env python3

import sys
import rospy
from example_service_pkg.srv import examplesrv

def service_client():
    rospy.init_node('service_client_node')
    if len(sys.argv) != 2:
        rospy.loginfo("Usage: service_client_node <input_string>")
        return

    rospy.wait_for_service('test_service')
    try:
        test_service = rospy.ServiceProxy('test_service', examplesrv)
        response = test_service(sys.argv[1])
        rospy.loginfo("Response from server: %s", response.out)
    except rospy.ServiceException as e:
        rospy.logerr("Failed to call service test_service: %s", str(e))

if __name__ == "__main__":
    service_client()

File: example_service_pkg/src/service_client.cpp
#include "ros/ros.h"
#include "example_service_pkg/examplesrv.h"

int main(int argc, char **argv) {
    ros::init(argc, argv, "service_client_node");

    if (argc != 2) {
        ROS_INFO("Usage: service_client_node <input_string>");
        return 1;
    }

    ros::NodeHandle n;
    ros::ServiceClient client = n.serviceClient<example_service_pkg::examplesrv>("test_service");
    example_service_pkg::examplesrv srv;
    srv.request.in = argv[1];

    if (client.call(srv)) {
        ROS_INFO("Response from server: %s", srv.response.out.c_str());
    } else {
        ROS_ERROR("Failed to call service test_service");
        return 1;
    }

    return 0;
}
